src/compiler/build/write-export-maps.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  isEligiblePrimaryPackageOutputTarget,
+  isOutputTargetDistCustomElements,
+  isOutputTargetDistLazyLoader,
+} from '@utils';
+import { relative } from '@utils';
+import { execSync } from 'child_process';
+
+import * as d from '../../declarations';
+import { PRIMARY_PACKAGE_TARGET_CONFIGS } from '../types/validate-primary-package-output-target';
+
+/**
+ * Create export map entry point definitions for the `package.json` file using the npm CLI.
+ * This will generate a root entry point for the package, as well as entry points for each component and
+ * the lazy loader (if applicable).
+ *
+ * @param config The validated Stencil config
+ * @param buildCtx The build context containing the components to generate export maps for
+ */
+export const writeExportMaps = (config: d.ValidatedConfig, buildCtx: d.BuildCtx) => {
+  const eligiblePrimaryTargets = config.outputTargets.filter(isEligiblePrimaryPackageOutputTarget);
+  if (eligiblePrimaryTargets.length > 0) {
+    const primaryTarget =
+      eligiblePrimaryTargets.find((o) => o.isPrimaryPackageOutputTarget) ?? eligiblePrimaryTargets[0];
+    const outputTargetConfig = PRIMARY_PACKAGE_TARGET_CONFIGS[primaryTarget.type];
+
+    if (outputTargetConfig.getModulePath) {
```

Comment:
Even though we have types for this but who knows how user may circumvent this
```suggestion
    if (typeof outputTargetConfig.getModulePath === 'function') {
```

Reply:
This is actually something we setup for primary output target validation. It's an internal config, not publicly exposed.

Reply:
no need for it then.